File: 0200-number-of-islands/0200-number-of-islands.go
func numIslands(grid [][]byte) int {
    islands := 0
    for i := range grid {
        for j := range grid[i] {
            curr := string(grid[i][j])
            if curr == "1" {
                islands += 1
                markLand(grid, i, j)
            }
            
        }
    }
    return islands
}

func markLand(grid [][]byte, i int, j int) {
    
    a := [][]int{
        {0, 1}, {1, 0}, {-1, 0}, {0, -1},
    }
    
    if i >= 0 && i < len(grid) && j >= 0 && j < len(grid[0]) && string(grid[i][j]) == "1" {
        // fmt.Println(grid[i][j])
        grid[i][j] = 50
        for _, y := range(a) {
        markLand(grid, i + y[0], j + y[1])
        }
    } else {
        return
    } 
    
    
    
     
}

File: 0200-number-of-islands/0200-number-of-islands.py
class Solution:      
            
    def numIslands(self, grid: List[List[str]]) -> int:
        islands = 0
        
        dir = [
            [1, 0],
            [-1, 0],
            [0, 1],
            [0, -1]
        ]
        # iterate through all islands, increase count by 1 when we encounter a 1 and change it to 2
        
        def markLand(grid, row, col):
            if (0 <= row < len(grid)) and (0 <= col < len(grid[0])) and grid[row][col] == "1":
                grid[row][col] = "2"
                for x, y in dir:
                    markLand(grid, row + x, col + y)
            else:
                return
                    
        for row in range(len(grid)):
            for col in range(len(grid[0])):
                if grid[row][col] == "1":
                    islands += 1
                    markLand( grid, row, col)
                    
        return islands
                    
        
                    
                


                